src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+        }
+        for (int i = 0; i < nShorts; i++) {
+            final int shift = i * 16 + srcPos;
+            dst[dstPos + i] = (short) (0xffff & (src >> shift));
+        }
+        return dst;
+    }
+
+    /**
+     * <p>
+     * Converts an int into an array of short using the default (little endian, Lsb0) byte and
+     * bit ordering.
+     * </p>
+     *
+     * @param src the int to convert
+     * @param srcPos the position in {@code src}, in bits, from where to start the conversion
+     * @param dst the destination array
+     * @param dstPos the position in {@code dst} where to copy the result
+     * @param nShorts the number of shorts to copy to {@code dst}, must be smaller or equal to
+     *            the width of the input (from srcPos to msb)
+     * @return {@code dst}
+     * @throws NullPointerException if {@code dst} is {@code null}
+     * @throws IllegalArgumentException if {@code (nShorts-1)*16+srcPos >= 32}
+     * @throws ArrayIndexOutOfBoundsException if {@code dstPos + nShorts > dst.length}
+     */
+    public static short[] intToShortArray(final int src, final int srcPos, final short[] dst, final int dstPos,
+            final int nShorts) {
+        if (0 == nShorts) {
+            return dst;
+        }
+        if ((nShorts - 1) * 16 + srcPos >= 32) {
+            throw new IllegalArgumentException("(nShorts-1)*16+srcPos is greater or equal to than 32");
+        }
+        for (int i = 0; i < nShorts; i++) {
+            final int shift = i * 16 + srcPos;
+            dst[dstPos + i] = (short) (0xffff & (src >> shift));
+        }
+        return dst;
+    }
+
+    /**
+     * <p>
+     * Converts a long into an array of byte using the default (little endian, Lsb0) byte and
+     * bit ordering.
+     * </p>
+     *
+     * @param src the long to convert
+     * @param srcPos the position in {@code src}, in bits, from where to start the conversion
+     * @param dst the destination array
+     * @param dstPos the position in {@code dst} where to copy the result
+     * @param nBytes the number of bytes to copy to {@code dst}, must be smaller or equal to the
+     *            width of the input (from srcPos to msb)
+     * @return {@code dst}
+     * @throws NullPointerException if {@code dst} is {@code null}
+     * @throws IllegalArgumentException if {@code (nBytes-1)*8+srcPos >= 64}
+     * @throws ArrayIndexOutOfBoundsException if {@code dstPos + nBytes > dst.length}
+     */
+    public static byte[] longToByteArray(final long src, final int srcPos, final byte[] dst, final int dstPos,
+            final int nBytes) {
+        if (0 == nBytes) {
+            return dst;
+        }
+        if ((nBytes - 1) * 8 + srcPos >= 64) {
+            throw new IllegalArgumentException("(nBytes-1)*8+srcPos is greater or equal to than 64");
+        }
+        for (int i = 0; i < nBytes; i++) {
+            final int shift = i * 8 + srcPos;
+            dst[dstPos + i] = (byte) (0xff & (src >> shift));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).